File: leetcode2/1easy/김대원/Q94.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> answer = new ArrayList<>();
        inOrder(root, answer);
        return answer;
    }

    public void inOrder(TreeNode node, List<Integer> answer) {
        if (node == null) return;
        inOrder(node.left, answer);
        answer.add(node.val);
        inOrder(node.right, answer);
    }
}

File: leetcode2/1easy/강병호/Q94.py
# Definition for a binary tree node.
# class 이재훈.TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Leetcode.이재훈.이재훈.Solution:
    def inorderTraversal(self, root: Optional[이재훈.TreeNode]) -> list[int]:
        result, stack = [], []
        curr = root
        while curr or stack:
            while curr:
                stack.append(curr)
                curr = curr.left
            curr = stack.pop()
            result.append(curr.val)
            curr = curr.right
        return result